soc/xtensa/espressif_esp32/esp32/Kconfig.defconfig.series

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_SERIES_ESP32
+
+config SOC_SERIES
+	default "esp32"
+
+config SOC_PART_NUMBER
+	default "esp32_d0wd_v3"		if SOC_ESP32_D0WD_V3
+	default "esp32_d0wdr2_v3"	if SOC_ESP32_D0WDR2_V3
+	default "esp32_u4wdh"		if SOC_ESP32_U4WDH
+	default "esp32_pico_v3"		if SOC_ESP32_PICO_V3
+	default "esp32_pico_v3_02"	if SOC_ESP32_PICO_V3_02
+	default "esp32_pico_d4"		if SOC_ESP32_PICO_D4
+	default "esp32_wroom_da_n4"	if SOC_ESP32_WROOM_DA_N4
+	default "esp32_wroom_da_n8"	if SOC_ESP32_WROOM_DA_N8
+	default "esp32_wroom_da_n16"	if SOC_ESP32_WROOM_DA_N16
+	default "esp32_wroom_32ue_n4"	if SOC_ESP32_WROOM_32UE_N4
+	default "esp32_wroom_32ue_n8"	if SOC_ESP32_WROOM_32UE_N8
+	default "esp32_wroom_32ue_n16"	if SOC_ESP32_WROOM_32UE_N16
+	default "esp32_wrover_e_n4r2"	if SOC_ESP32_WROVER_E_N4R2
+	default "esp32_wrover_e_n8r2"	if SOC_ESP32_WROVER_E_N8R2
+	default "esp32_wrover_e_n16r2"	if SOC_ESP32_WROVER_E_N16R2
+	default "esp32_wrover_e_n4r8"	if SOC_ESP32_WROVER_E_N4R8
+	default "esp32_wrover_e_n8r8"	if SOC_ESP32_WROVER_E_N8R8
+	default "esp32_wrover_e_n16r8"	if SOC_ESP32_WROVER_E_N16R8
```

Comment:
Well, the modules `esp32_wroom`, `esp32_wrover` etc. have some substantial differences, like flash/psram size and also the GPIO availability (which is mostly handled by hal/IDF). And regarding the mixing modules with SOCs, I took inspiration from the already supported SOC/SIP. For example, `NRF9160` or `NRF9161` is listed the same way as other SOCs. 

Overall, this list of SOC/SIP comes from the ESP32 product selector, and I choose the up-to-date modules/soc or modules/soc which are already used by the board supported in Zephyr. Also, the list is meant to serve as a reference for any new board designers.

Reply:
> Well, the modules esp32_wroom, esp32_wrover etc. have some substantial differences, like flash/psram size and also the GPIO availability (which is mostly handled by hal/IDF).

In added dtsi files I see only flash/ram differences. Also, as an example, `esp32s2_mini_n4r2.dtsi`, `esp32s2_solo_n4r2.dtsi` and `esp32s2_wrover_n4r2.dtsi` are all the same. So is it worth introducing all of them? With all added features, there is a ratio of gain and increased maintenance. I see that those can be used by downstream boards, but is it enough to justify adding so many module variants and maintaining them in Zephyr upstream?

There also comes the question where to put those files, maybe it would be good to create some "module" directory to group all of them and separate them from SoC definitions.

> And regarding the mixing modules with SOCs, I took inspiration from the already supported SOC/SIP. For example, NRF9160 or NRF9161 is listed the same way as other SOCs.

nRF9160/nRF9161 is a bit different, as those can be obtained only in SiP, never in smaller entities. Each of them is probably based on different SoC, since they have different capabilities. ESP32 modules are something built on top of SoCs, they are clearly something between an SoC and a board (PCB of a ready product or devkit).

Can we move it to another PR, so we can progress with refactoring/bugfixes regardless of what we decide to do with modules?

Reply:
Regarding the SOC/SIP granularity - I created only those modules which have different properties. The `FLASH` size, `PSRAM` size, `GPIO` count, and the `certification` status were taken into account when making the selection. The selection wasn't random or bulk. Currently, not all properties are handled in dtsi, but will be added later in another PR (GPIO count).

Regarding the SIP vs SOC - SOCs used inside the SIPs are often different from the plain SOC by how they incorporate FLASH or PSRAM memories, which are often integrated into the same package/die, or how many GPIOs they offer for users. So the difference between SIP and SOC is from very blurry to nonexistent. So comparing `ESP32xx` modules to `NRF916x` is spot on.